src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -111,8 +111,21 @@ private CompletionProvider GetCompletionProvider(CompletionItem item)
                     var mappedSpan = triggerSnapshotSpan.TranslateTo(
                         this.SubjectBuffer.CurrentSnapshot, SpanTrackingMode.EdgeInclusive);
 
+                    // If the client is providing a caret position, then it's likely they're
+                    // making a large/complex change to the document.  In this case try to
+                    // get the editor to actually apply this as a minimal-change so that the
+                    // document change looks nice and tidy.  This won't mess anything up as we
+                    // will then place the caret manually.
+                    //
+                    // If the caret position is not provided, then we do not want a minimal-change
```

Comment:
Ok.  I've made the discussed change and have added a test to validate the behavior is as we want it to be.